lib/decompress/zstd_decompress.c

```diff
@@ -544,6 +556,59 @@ static size_t readSkippableFrameSize(void const* src, size_t srcSize)
     }
 }
 
+/*! ZSTD_findFrameSize() :
+ * Returns the raw size of the first frame in a buffer.  Advancing the pointer by this amount will move to the next frame.
+ *  `src` should point to the start of a buffer with one or more ZSTD encoded frames.
+ *  `srcSize` must be at least as large as the frame header.
+ *            hint : any size >= `ZSTD_frameHeaderSize_max` is large enough.
+ *  @return : - size of `src` of the first frame, if known.  Advancing the pointer by this amount will move to the next frame.
+ *            - ZSTD_CONTENTSIZE_UNKNOWN if the size cannot be determined
+ *            - ZSTD_CONTENTSIZE_ERROR if an error occurred (e.g. invalid magic number, srcSize too small)
+ */
+unsigned long long ZSTD_findFrameSize(const void *src, size_t srcSize)
+{
+    ZSTD_frameHeader zfh;
+    if (ZSTD_getFrameHeader(&zfh, src, srcSize) != 0)
+        return ZSTD_CONTENTSIZE_ERROR;
+    if (zfh.frameType == ZSTD_skippableFrame) {
+        return readSkippableFrameSize(src, srcSize);
+    } else {
+        return ZSTD_findFrameCompressedSize(src, srcSize);
```

Comment:
Isn't `ZSTD_findFrameCompressedSize()` already able to provide the size of a skippable frame ?

Reply:
Ah, I thought that was only for compressed frames; removed the new function.